nominal/cli/util/click_log_handler.py

```diff
@@ -0,0 +1,74 @@
+from __future__ import annotations
+
+import logging
+import typing
+
+import click
+
+
+class ClickLogHandler(logging.StreamHandler[typing.TextIO]):
+    """Logging stream handler that routes and styles log messages through click
+    instead of directly routing to stderr.
+
+    This has several advantages, namely, differing log levels can be printed differently
+    to provide additional visual amplification of errors and warnings. Furthermore,
+    there are numerous improvements to support within windows.
+
+    See: https://click.palletsprojects.com/en/8.1.x/utils/#printing-to-stdout
+    """
+
+    LEVEL_TO_COLOR_MAP = {
+        logging.DEBUG: "blue",
+        logging.INFO: "cyan",
+        logging.WARNING: "yellow",
+        logging.ERROR: "red",
+        logging.FATAL: "bright_red",
+    }
+
+    def __init__(self, stream: typing.TextIO | None = None, no_color: bool = False):
+        """Instantiate a ClickLogHandler
+
+        Args:
+            stream: TextIO stream to pipe filtered and rendered log messages to
+            no_color: If True, don't colorize/style log messages by level during rendering
+        """
+        if stream is None:
+            stream = click.get_text_stream("stderr")
+
+        super().__init__(stream)
+
+        self._no_color = no_color
+
+    def emit(self, record: logging.LogRecord) -> None:
+        """Render a log record into a stylized string and print to the configured TextIO
+        output stream of the log handler.
+        """
+        try:
+            msg = self.format(record)
+            if not self._no_color:
+                msg = click.style(msg, fg=self.LEVEL_TO_COLOR_MAP.get(record.levelno, "white"))
+
+            stream = self.stream
+            # issue 35046: merged two stream.writes into one.
+            stream.write(msg + self.terminator)
+            self.flush()
+        except RecursionError:  # See issue 36272
```

Comment:
can you link the issues in the comments?

Reply:
maybe this should be handled in `format()` rather than `emit()`? then you can just super().format() and optionally change the color